app/components/primer/heading_component.rb

```diff
@@ -1,23 +1,46 @@
 # frozen_string_literal: true
 
 module Primer
-  # Use the Heading component to wrap a component that will create a heading element
+  # Use Heading to communicate page organization and hierarchy.
+  #
+  # ## Usage
+  #
+  # - Use as the title of a section or sub section. 
+  # - Use one `<h1>`` per page. 
+  # ### Do not
+  # - Skip heading levels. For instance, do not jump from `<h1>` to `<h3>`.
+  # - Use Heading for sole stylistic purposes. For styling text, consider using [Text] with relevant
+  # [typography arguments].
+  #
+  # [Text]: https://primer.style/view-components/components/text
+  # [typography arguments]: https://primer.style/view-components/system-arguments#typography
+  #
+  # ## Accessibility
+  # - Headings are used by assistive technologies to navigate a page. 
+
   class HeadingComponent < Primer::Component
     status :beta
 
+    TAG_OPTIONS = [:h1, :h2, :h3, :h4, :h5, :h6].freeze
+
     # @example Default
-    #   <%= render(Primer::HeadingComponent.new) { "H1 Text" } %>
+    #   <%= render(Primer::HeadingComponent.new(tag: :h1)) { "H1 Text" } %>
     #   <%= render(Primer::HeadingComponent.new(tag: :h2)) { "H2 Text" } %>
     #   <%= render(Primer::HeadingComponent.new(tag: :h3)) { "H3 Text" } %>
     #
+    # @param tag [String]  <%= one_of(Primer::HeadingComponent::TAG_OPTIONS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-    def initialize(**system_arguments)
+    def initialize(tag:, **system_arguments)
       @system_arguments = system_arguments
-      @system_arguments[:tag] ||= :h1
+      @system_arguments[:tag] = tag
     end
 
     def call
       render(Primer::BaseComponent.new(**@system_arguments)) { content }
     end
+
+    def before_render
+      raise ArgumentError, "Tag must be set to one of #{TAG_OPTIONS.join(',')}" unless TAG_OPTIONS.include?(@system_arguments[:tag])
```

Comment:
I'd love to see us abstract this into something similar to `fetch_or_fallback`, what do you think?

Reply:
I noticed that method and some unless Rails.env.production?` checks and wanted to get some insight. Why do we need to provide a graceful fallback if the env is production in some cases?

Reply:
🤔 good question. I'm not sure how we'd handle this case here, but the reason we fallback in production is to prevent component misuse from leading to 500 errors. Do you think we should 500 for passing an incorrect tag option?

Reply:
I see. How does a component misuse get into production? I think I would have expected an error to be raised at dev stage that must be fixed. Or.. maybe this happens when an API changes later on and existing usage isn't corrected?

Reply:
Some of our components have restricted tags defined by `TAG_OPTIONS` and some don't. I think we've discussed that eventually we'll want to define allowed tags for all components and it seems like something that's [on the radar ](https://github.com/github/primer/issues/115)of PRC team as well!

Maybe we can have the tag validation check called in `initialize` of `BaseComponent` like:

```.rb
    def validate_tags(tag)
      return unless TAG_OPTIONS.present? # Remove once allowed tags are clearly defined for all components. 
      
      raise ArgumentError, "tag must be set to one of #{TAG_OPTIONS.join(', ')}" unless TAG_OPTIONS.include?(tag)
    end
```

But this depends on whether it's important to fail gracefully for incorrect tags and I don't really have context.
Is there a reason to provide this fallback for tags in production, or is it safe to fail loudly? cc: @manuelpuyol

Reply:
The most common route is untested code. Someone makes a change, it's not covered by a test, and they don't run it locally or in review lab. It's not ideal, but we've run into it before.

Reply:
> How does a component misuse get into production? 

This does happen on big migrations (especially when we add a breaking change to a component). Heading may not be widely used, but we have some components, such as `Subhead` that are used hundreds of times.
When migrating all of those component calls to use a new API, you may miss 1 or 2 in untested places, which could lead to 500s

I think it is safer to not raise in production simply because we are quickly changing our components and would lead us to less headaches 😅

Reply:
I see, thanks for providing some context! 

I'll remove the default but keep a fallback to handle invalid tags.